mypy/messages.py

```diff
@@ -819,6 +819,21 @@ def incompatible_typevar_value(self, callee: CallableType, index: int,
         self.fail('Type argument {} of {} has incompatible value {}'.format(
             index, callable_name(callee), self.format(type)), context)
 
+    def incompatible_constrained_arguments(self,
+                                           callee: CallableType,
+                                           index: int,
+                                           constraints: Mapping[str, Sequence[str]],
+                                           context: Context) -> None:
+        for key, values in constraints.items():
+            self.fail('Type argument {} of {} has incompatible value'.format(
+                index, callable_name(callee)), context)
+            if len(values) == 2:
+                constraint_str = '{} or {}'.format(values[0], values[1])
+            elif len(values) > 3:
+                constraint_str = ', '.join(values[:-1]) + ', or ' + values[-1]
+            self.note('"{}" must be all one type: {}'.format(
```

Comment:
OK, then again we could postpone this to the next PR (probably the idea is to use ``types`` argument for ``apply_generic_arguments`` and ``map_actuals_to_formals`` from ``checkexpr.py``).